supportedBackends/spark/src/main/scala/cromwell/backend/impl/spark/SparkProcess.scala

```diff
@@ -0,0 +1,93 @@
+package cromwell.backend.impl.spark
+
+import java.nio.file.Path
+
+import com.typesafe.scalalogging.StrictLogging
+import cromwell.core.{TailedWriter, UntailedWriter}
+import cromwell.core.PathFactory.EnhancedPath
+
+import scala.sys.process._
+import better.files._
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+
+object SparkCommands {
+  val SPARK_SUBMIT = "spark-submit"
+  val SEPARATOR = "--"
+  val EXECUTOR_CORES = "total-executor-cores"
+  val EXECUTOR_MEMORY = "executor-memory"
+  val APP_MAIN_CLASS = "class"
+  val MASTER = "master"
+  val DEPLOY_MODE = "deploy-mode"
+  val SPARK_APP_WITH_ARGS = "spark_app_with_args"
+  val MEMORY_UNIT = "G" //Accepts only GB
+  val SHELL_CMD = "sh"
+  val PIPELINE = "&>"
+}
+
+class SparkCommands extends StrictLogging {
+  import SparkCommands._
+  import SparkClusterConstants._
+  /**
+    * Writes the script file containing the user's command from the WDL as well
+    * as some extra shell code for monitoring jobs
+    */
+  def writeScript(instantiatedCommand: String, filePath: Path, containerRoot: Path) = {
+    filePath.write(
+      s"""#!/bin/sh
+          |cd $containerRoot
+          |$instantiatedCommand
+          |echo $$? > rc
+          |""".stripMargin)
+  }
+
+  def sparkSubmitCommand(attributes: Map[String, Any]): String = {
+    val stringBuilder = StringBuilder.newBuilder
+    val commandPlaceHolder = " %s%s %s "
+    val sparkHome = Try(sys.env("SPARK_HOME")) match {
+      case Success(s) => Some(s)
+      case Failure(ex) => logger.warn(s"Spark home does not exist picking up default command")
+        None
+    }
+
+    val sparkSubmit = sparkHome
+      .map(p => String.format("%s%s", p, s"/bin/$SPARK_SUBMIT"))
+      .getOrElse(SPARK_SUBMIT)
+
+    attributes.foreach {
+      case (key, _) => attributes.get(key) match {
+        case Some (v) => commandPlaceHolder.format (SEPARATOR, key, v)
```

Comment:
Hmm, I don't think this is a functional argument? I think it's just a function call with an accidental space before the `(` - So, `.format(...)`